tests/backup/backup_helper.go

```diff
@@ -1629,6 +1667,310 @@ func restoreSuccessWithReplacePolicy(restoreName string, orgID string, retryDura
+			errors = append(errors, err)
+			continue
+		}
+		for _, restoredVol := range actualRestoredVolumes {
+			actualVolumeMap[restoredVol.ID] = restoredVol
+		}
+		log.Infof("volumes bounded to the PVCs in the context [%s] are [%+v]", expectedRestoredAppContextNamespace, actualRestoredVolumes)
+
+		// looping over the list of volumes that PX-Backup says it restored, to run some checks
+		for _, restoredVolInfo := range apparentlyRestoredVolumes {
+			if namespaceMappings[restoredVolInfo.SourceNamespace] == expectedRestoredAppContextNamespace {
+				if restoredVolInfo.Status.Status != api.RestoreInfo_StatusInfo_Success /*Can this also be partialsuccess?*/ {
+					err := fmt.Errorf("in restore [%s], the status of the restored volume [%s] was not Success. It was [%s] with reason [%s]", restoreName, restoredVolInfo.RestoreVolume, restoredVolInfo.Status.Status, restoredVolInfo.Status.Reason)
+					errors = append(errors, err)
+					continue
+				}
+
+				var actualVol *volume.Volume
+				var ok bool
+				if actualVol, ok = actualVolumeMap[restoredVolInfo.RestoreVolume]; !ok {
+					err := fmt.Errorf("in restore [%s], said restored volume [%s] cannot be found in the actual cluster [%s]", restoreName, restoredVolInfo.RestoreVolume, expectedRestoredAppContextNamespace)
+					errors = append(errors, err)
+					continue
+				}
+
+				if actualVol.Name != restoredVolInfo.Pvc {
+					err := fmt.Errorf("in restore [%s], for restored volume [%s], PVC used is given as [%s], but actual PVC used was found to be [%s]", restoreName, restoredVolInfo.RestoreVolume, restoredVolInfo.Pvc, actualVol.Name)
+					errors = append(errors, err)
+				}
+
+				var expectedVolumeDriver string
+				if os.Getenv("BACKUP_TYPE") == "Generic" {
+					expectedVolumeDriver = "kdmp"
+				} else {
+					expectedVolumeDriver = Inst().V.String()
+				}
+
+				if restoredVolInfo.DriverName != expectedVolumeDriver {
+					err := fmt.Errorf("in restore [%s], for restored volume [%s], volume driver actually used is given as [%s], but expected is [%s]", restoreName, restoredVolInfo.RestoreVolume, restoredVolInfo.DriverName, expectedVolumeDriver)
+					errors = append(errors, err)
+				}
+			}
+		}
+
+		// VALIDATE APPLICATIONS
+		log.InfoD("Validate applications in restored namespace [%s] due to restore [%s]", expectedRestoredAppContextNamespace, restoreName)
+		errorChan := make(chan error, errorChannelSize)
+		ValidateContext(expectedRestoredAppContext, &errorChan)
+		for err := range errorChan {
+			errors = append(errors, err)
+		}
+	}
+
+	errStrings := make([]string, 0)
+	for _, err := range errors {
+		if err != nil {
+			errStrings = append(errStrings, err.Error())
+		}
+	}
+
+	if len(errStrings) > 0 {
+		return fmt.Errorf("ValidateRestore Errors: {%s}", strings.Join(errStrings, "}\n{"))
+	} else {
+		return nil
+	}
```

Comment:
How about we return `[]error` and convert `[]error` to a string for logging outside? @mkoppal-px

Reply:
That is what I did originally. But Mithun suggested that we just return just one error. I feel like that makes sense in this case as _any_ error _is_ an error in this function and should be treated as such.

Reply:
Yeah, I think it is better if we can create a new function called log.FailOnMultipleErrors or something similar. This function would take an array slice and print each error along with its corresponding index.

Reply:
While this might be useful, returning a single error has the advantage of calling this in other functions that return a single error. So, at least for now, let's leave it. 

Though, technically, given that this is a "part of testing", we should directly do `log.Errorf` wherever there's an error (and continue) and then we can set flags that we can use to determine if we want to fail due to error in validation (as validation is not an obstruction in testing for other errors in the same testcase, before ending the whole thing).

for now, let it be thus